pkg/console/controllers/route/controller.go

```diff
@@ -92,27 +81,25 @@ func NewRouteSyncController(
 		targetNamespace:      targetNamespace,
 		routeName:            routeName,
 		// events
-		recorder:     recorder,
-		cachesToSync: nil,
-		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
-		ctx:          ctx,
+		resourceSyncer: resourceSyncer,
 	}
 
-	operatorClient.Informer().AddEventHandler(ctrl.newEventHandler())
-	operatorConfigInformer.Informer().AddEventHandler(ctrl.newEventHandler())
-	routeInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	configNameFilter := util.NamesFilter(api.ConfigResourceName)
+	targetNameFilter := util.NamesFilter(api.OpenShiftConsoleName)
 
-	ctrl.cachesToSync = append(ctrl.cachesToSync,
-		operatorClient.Informer().HasSynced,
-		operatorConfigInformer.Informer().HasSynced,
-		routeInformer.Informer().HasSynced,
-	)
-
-	return ctrl
+	return factory.New().
+		WithFilteredEventsInformers( // configs
+			configNameFilter,
+			operatorConfigInformer.Informer(),
+		).WithFilteredEventsInformers( // console resources
```

Comment:
shouldn't you react to changes of the CMs and secrets, too?

Reply:
yes, but not the Secrets name, since those names are arbitrary 👍